includes/discovery/sensors/fanspeeds/mikrotik.inc.php

```diff
@@ -1,28 +1,19 @@
 <?php
 
 if ($device['os'] == 'routeros') {
-    $oids = snmp_walk($device, 'mtxrHlActiveFan', '-OsqnU', 'MIKROTIK-MIB');
-    d_echo($oids."\n");
+    $descr_prefix = 'fan ';
+    $oids = array(
+        '.1.3.6.1.4.1.14988.1.1.3.17.0', // MIKROTIK-MIB::mtxrHlFanSpeed1.0
+        '.1.3.6.1.4.1.14988.1.1.3.18.0', // MIKROTIK-MIB::mtxrHlFanSpeed2.0
+    );
 
-    if ($oids !== false) {
-        echo 'MIKROTIK-MIB ';
-    }
-
-    $divisor = 1;
-    $type    = 'mikrotik';
+    echo 'MIKROTIK-MIB ';
+    foreach ($oids as $index => $oid) {
+        $value = trim(snmp_get($device, $oid, '-Oqv'), '"');
 
-    foreach (explode("\n", $oids) as $data) {
-        $data = trim($data);
-        if ($data) {
-            list($oid,$descr) = explode(' ', $data, 2);
-            $split_oid        = explode('.', $oid);
-            $index            = $split_oid[(count($split_oid) - 1)];
-            $descr            = 'Fan '.$index;
-            $oid              = '.1.3.6.1.4.1.14988.1.1.3.9.'.$index;
-            $fanspeed         = (snmp_get($device, $oid, '-Oqv') / $divisor);
-            if ($fanspeed > 0) {
-                discover_sensor($valid['sensor'], 'fanspeed', $device, $oid, $index, $type, $descr, $divisor, '1', null, null, null, null, $fanspeed);
-            }
+        if (is_numeric($value)) {
+            $descr = $descr_prefix . ($index + 1);
+            discover_sensor($valid['sensor'], 'fanspeed', $device, $oid, $index, 'snmp', $descr, 1, 1, null, null, null, null, $value);
```

Comment:
The old OID actually asks for which fan "stage" is actually being in use, whereas the new OIDs for fans actually give the RPM in an INTEGER format.

Old was a STRING format that just said "main" or "backup" in regards to which fan stage was actually being used. Useful as well but not as useful as the RPM ratings. I believe other devices also did the same as well that have supported this.